internal/proto/internal.proto

```diff
@@ -67,4 +67,48 @@ message PrContents {
 
   minder.v1.PullRequest pr = 1;
   repeated File files = 2;
-}
\ No newline at end of file
+}
+
+message SelectorProvider {
+  // the name of the provider, e.g. github-app-jakubtestorg
+  string name = 1;
+  // the class of the provider, e.g. github-app
+  string class = 2;
+}
+
+message SelectorRepository {
+  // the full name of the repository, e.g. stacklok/minder
+  string name = 1;
+  // the provider of the repository
+  SelectorProvider provider = 2;
+
+  // is_fork is true if the repository is a fork, nil if "don't know" or rather
+  // not applicable to this provider
+  optional bool is_fork = 3;
+  // is_private is true if the repository is private, nil if "don't know" or rather
+  // not applicable to this provider
+  optional bool is_private = 4;
+}
+
+message SelectorArtifact {
+  // the full name of the artifact, e.g. stacklok/minder-server
+  string name = 1;
+  // the provider of the artifact
+  SelectorProvider provider = 2;
+
+  // the type of the artifact, e.g. "container"
+  string type = 4;
+}
+
+message SelectorEntity {
+  // one of repository, pull_request, artifact (see oneof entity)
+  minder.v1.Entity entity_type = 1;
+  // the name of the entity, same as the name in the entity message
+  string name = 2;
+  SelectorProvider provider = 3;
+
+  oneof entity {
+    SelectorRepository repository = 4;
+    SelectorArtifact artifact = 5;
+  }
+}
```

Comment:
This is an internal implementation detail that currently makes it easier to implement CEL, right?

I'm concerned over time that encoding all the properties into a protocol buffer in core Minder will make it harder for new providers to add properties that make sense, but we can cross that bridge when we get there.  (The alternative is that it's easy for providers to add new properties, but little uniformity across providers -- this pushes us on the higher-friction-but-more-commonality path, which might even be right.)

Reply:
> This is an internal implementation detail that currently makes it easier to implement CEL, right?

yes

> 
> I'm concerned over time that encoding all the properties into a protocol buffer in core Minder will make it harder for new providers to add properties that make sense, but we can cross that bridge when we get there. (The alternative is that it's easy for providers to add new properties, but little uniformity across providers -- this pushes us on the higher-friction-but-more-commonality path, which might even be right.)

Right, I hear you. The properties that are expressed as attributes are only those that those entities will most commonly have (names, is_fork/is_private and those are optional bool, too). I expect any provider-specific attributes to be just key-value pairs in a map.

We could even go with just a map, but the nice thing about passing proto messages to CEL is that you CEL can check [if the attributes exist](https://github.com/jhrozek/minder/blob/35e207a748d5e1239c9e1cbee2c917e2476778cb/internal/engine/selectors/selectors_test.go#L265) or [warn about mismatch between entity types and the attribute](https://github.com/jhrozek/minder/blob/35e207a748d5e1239c9e1cbee2c917e2476778cb/internal/engine/selectors/selectors_test.go#L253). I think it would be nice to surface these to the user when they create the selectors.

We won't have those for a generic map.